src/TensorOneD.hpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2015 - 2023, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef TENSORONED_HPP_INCLUDE
+#define TENSORONED_HPP_INCLUDE
+
+#include <cstdint>
```

Comment:
AFAICT, no special int types are used.
stdio.h is needed for the size_t definition.
```suggestion
#include <stdio.h>
```